diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,260 @@
+            CLASS, FUN, PROPERTY -> {
+                checkBlankLineAfterKdoc(node, EOL_COMMENT)
+                checkBlankLineAfterKdoc(node, KDOC)
+                checkBlankLineAfterKdoc(node, BLOCK_COMMENT)
+            }
+            IF -> {
+                handleIfElse(node)
+            }
+            EOL_COMMENT -> {
+                handleEolComments(node, configuration)
+            }
+            BLOCK_COMMENT -> {
+                handleBlockComments(node, configuration)
+            }
+            KDOC -> {
+                handleKdocComments(node)
+            }
+        }
+    }
+
+    private fun checkBlankLineAfterKdoc(node: ASTNode, type: IElementType) {
+        val kdoc = node.getFirstChildWithType(type)
+        val nodeAfterKdoc = kdoc?.treeNext
+        if (nodeAfterKdoc?.elementType == ElementType.WHITE_SPACE && nodeAfterKdoc.numNewLines() > 1) {
+            WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, kdoc.text, nodeAfterKdoc.startOffset) {
+                nodeAfterKdoc.leaveOnlyOneNewLine()
+            }
+        }
+    }
+
+    private fun handleKdocComments(node: ASTNode) {
+        if (node.treeParent.treeParent.elementType == BLOCK) {
+            checkCommentsInCodeBlocks(node.treeParent) // node.treeParent is a node that contains a comment.
+        } else if (node.treeParent.elementType != IF){
+            checkClassComment(node)
+        }
+    }
+
+    private fun handleEolComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        checkSpaceBetweenPropertyAndComment(node, configuration)
+
+        if (node.treeParent.elementType == BLOCK && node.treeNext != null) {
+            checkCommentsInCodeBlocks(node)
+        } else if (node.treeParent.lastChildNode != node && node.treeParent.elementType != IF) {
+            checkClassComment(node)
+        }
+
+        checkWhiteSpaceBeforeComment(node)
+    }
+
+    private fun handleBlockComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        checkSpaceBetweenPropertyAndComment(node, configuration)
+
+        if (node.treeParent.elementType == BLOCK && node.treeNext != null) {
+            checkCommentsInCodeBlocks(node)
+        } else if (node.treeParent.lastChildNode != node && node.treeParent.elementType != IF) {
+            checkClassComment(node)
+        }
+    }
+
+    private fun handleIfElse(node: ASTNode) {
+        if(node.hasChildOfType(ELSE)) {
+            val elseKeyWord = node.getFirstChildWithType(ELSE_KEYWORD)!!
+            if(elseKeyWord.treePrev.treePrev.elementType.let { it == EOL_COMMENT ||
+                            it == KDOC || it == BLOCK_COMMENT }) {
+                IF_ELSE_COMMENTS.warnAndFix(configRules, emitWarn, isFixMode, elseKeyWord.treePrev.treePrev.text, node.startOffset) {
+                    val elseBlock = node.getFirstChildWithType(ELSE)!!
+                    if(elseBlock.hasChildOfType(BLOCK)) {
```

Comment:
Maybe extract `elseBlock.getFirstChildWithType(BLOCK)!!` to a variable? It is used four times in four consecutive lines